File: Aggregate.php
<?php

// Set current directory
$cd = dirname(__FILE__);

$handle = fopen($cd."/DATA/BigCompaniesData.csv", "r");    
    
$row = 0;
$values = [];

// READ CSV DATA
while (($data = fgetcsv($handle, 1000, ",")) !== FALSE) {        
    /* SKIP HEADER */
    if($row == 0) { $row++; continue; }
    
    /* OBTAIN VALUES */
    $values[$row]['Company'] = $data[0];
    $values[$row]['Industry'] = $data[1];
    $values[$row]['Revenues'] = $data[2];
    $values[$row]['Assets'] = $data[3];
    $values[$row]['Equity'] = $data[4];
    $values[$row]['NetIncome'] = $data[5];
    $values[$row]['StockPrice'] = $data[6];
    $values[$row]['Employees'] = $data[7];
    
    $row++;
}

// DEFINE DISTINCT INDUSTRIES ARRAY
$industries = [];
for ($i=1; $i<=sizeof($values); $i++) {
    $industries[]  = $values[$i]['Industry'];
}
$industries = array_unique($industries, SORT_REGULAR);
sort($industries);


// DEFINE AGGREGATE DATA ARRAY
$aggdata = [];
foreach ($industries as $i) {
    $ind = str_replace(' ', '', $i);
    $aggdata[$ind]['Industry'] = 0;
    $aggdata[$ind]['SumOfRevenues'] = 0;
    $aggdata[$ind]['AvgOfAssets'] = 0;
    $aggdata[$ind]['AvgOfEquity'] = 0;
    $aggdata[$ind]['MaxOfNetIncome'] = 0;
    $aggdata[$ind]['MinOfNetIncome'] = 1000000000000;
    $aggdata[$ind]['AvgOfStockPrice'] = 0;
    $aggdata[$ind]['SumOfEmployees'] = 0;
}


// LOOP THROUGH DATA CONDITIONAL ON INDUSTRY
foreach ($industries as $i) {
    $cnt = 0;
    $ind = str_replace(' ', '', $i);    
    for ($j=1; $j<=sizeof($values); $j++) {        
           
        if ($values[$j]['Industry'] == $i) {
            
            $aggdata[$ind]['Industry'] = $i;
            $aggdata[$ind]['SumOfRevenues'] = $aggdata[$ind]['SumOfRevenues'] + $values[$j]['Revenues'];              
            $aggdata[$ind]['AvgOfAssets'] = $aggdata[$ind]['AvgOfAssets'] + $values[$j]['Assets'];                
            $aggdata[$ind]['AvgOfEquity'] = $aggdata[$ind]['AvgOfEquity'] + $values[$j]['Equity'];            
            $aggdata[$ind]['MaxOfNetIncome'] = floatval(max($aggdata[$ind]['MaxOfNetIncome'], $values[$j]['NetIncome']));            
            $aggdata[$ind]['MinOfNetIncome'] = floatval(min($aggdata[$ind]['MinOfNetIncome'], $values[$j]['NetIncome']));            
            $aggdata[$ind]['AvgOfStockPrice'] = $aggdata[$ind]['AvgOfStockPrice'] + $values[$j]['StockPrice'];            
            $aggdata[$ind]['SumOfEmployees'] = $aggdata[$ind]['SumOfEmployees'] + $values[$j]['Employees'];           
            $cnt++;
            
        }

    }
    $aggdata[$ind]['AvgOfAssets'] = $aggdata[$ind]['AvgOfAssets'] / ($cnt);
    $aggdata[$ind]['AvgOfEquity'] = $aggdata[$ind]['AvgOfEquity'] / ($cnt);
    $aggdata[$ind]['AvgOfStockPrice'] = $aggdata[$ind]['AvgOfStockPrice'] / ($cnt);
}


// OUTPUT TO CSV
$fp = fopen($cd.'/AggregatedData_php.csv', 'w');

/* COLUMN HEADERS */
fputcsv($fp, array_keys($aggdata['Banks']));

/* DATA ROWS */
foreach ($industries as $i) {
    $ind = str_replace(' ', '', $i);
    fputcsv($fp, $aggdata[$ind]);
}
fclose($fp);

echo "Successfully aggregated data and outputted to CSV file!\n";

?>

File: Aggregate.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Set;
import java.util.HashSet;
import java.util.Collections;

import java.lang.Math;
 
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.BufferedReader;
import java.io.IOException;

class DataObj {
      
    String compfld; String indfld; 
    float revfld; float assetfld; float equityfld;
    float netincfld; float stockfld; float empfld;    
  
    public DataObj(String compfld, String indfld, float revfld, float assetfld,
                        float equityfld, float netincfld, float stockfld, float empfld) {
      
            this.compfld = compfld;
	    this.indfld = indfld;
	    this.revfld = revfld;
	    this.assetfld = assetfld;
	    this.equityfld = equityfld;
	    this.netincfld = netincfld;
            this.stockfld = stockfld;
            this.empfld = empfld;
    }
  
}

public class Aggregate {

      public static void main(String[] args) {
	
            String currentDir = new File("").getAbsolutePath();
            ArrayList<String> industrylist = new ArrayList<String>();
            Set<String> hs = new HashSet<>();            
            ArrayList<DataObj> datalist = new ArrayList<DataObj>();            
                        
            String csvFile = currentDir + "\\Data\\" + "BigCompaniesData.csv";
            BufferedReader br = null;
            String line = "";
            String csvSplitBy = ",";
                
            int row = 0;
            try {
                  br = new BufferedReader(new FileReader(csvFile));
                  while ((line = br.readLine()) != null) {
                        // Skip column headers
                        if (row==0){ row++; continue;}                        	   
                        
                        // Split by comma separator
                        line = line.replace(", Inc.", " Inc.");
                        String[] doodle = line.split(csvSplitBy);
                        
                        // Add into Aggregate Object
                        datalist.add(new DataObj(doodle[0], doodle[1],
                                                Float.valueOf(doodle[2]), Float.valueOf(doodle[3]), Float.valueOf(doodle[4]),
                                                Float.valueOf(doodle[5]), Float.valueOf(doodle[6]), Float.valueOf(doodle[7])));
                        
                        industrylist.add(doodle[1]);
                  }
                
                  // remove duplicates and alpha order
                  hs.addAll(industrylist);
                  industrylist.clear();
                  industrylist.addAll(hs);
                  Collections.sort(industrylist);
                
                  FileWriter writer = new FileWriter(currentDir + "\\AggregatedData_java.csv");
                  writer.append("Industry,SumOfRevenues,AvgOfAssets,AvgOfEquity,MaxOfNetIncome,MinOfNetIncome,AvgOfStockPrice,SumOfEmployees\n");
          
                  for(String ind: industrylist) {                        
                        String csvline = industryAggergation(ind, datalist);
                        writer.append(csvline);
                  }
                  writer.flush();
                  writer.close();
                  System.out.println("Successfully aggregated data and outputted to CSV!");
    
            } catch (FileNotFoundException e) {
                    e.printStackTrace();
            } catch (IOException e) {
                    e.printStackTrace();	
            }
        
      }
      
      public static String industryAggergation(String indtype, ArrayList<DataObj> rawdata) {
            
            String Industry = ""; 
            float SumOfRevenues = 0; float AvgOfAssets = 0; float AvgOfEquity = 0; float MaxOfNetIncome = 0;
            double MinOfNetIncome = 1000000000000L; float AvgOfStockPrice =0; float SumOfEmployees = 0;
            int cnt = 0;  
            
            cnt = 0;                       
            for(DataObj a : rawdata) {
                  
                  if (a.indfld.equals(indtype)) {                                  
                        Industry = indtype;
                        SumOfRevenues = SumOfRevenues + a.revfld;
                        AvgOfAssets = AvgOfAssets + a.assetfld;
                        AvgOfEquity = AvgOfEquity + a.equityfld;
                        MaxOfNetIncome = Math.max(MaxOfNetIncome, a.netincfld);
                        MinOfNetIncome = Math.min(MinOfNetIncome, a.netincfld);
                        AvgOfStockPrice = AvgOfStockPrice + a.stockfld;
                        SumOfEmployees  = SumOfEmployees + a.empfld;
                        cnt = cnt + 1;
                  } else {                                    
                        continue;
                  }
                  
            }
            return Industry + "," + SumOfRevenues + "," + AvgOfAssets/cnt + "," + AvgOfEquity/cnt + "," +
                   MaxOfNetIncome + "," + MinOfNetIncome + "," + AvgOfStockPrice/cnt + "," + SumOfEmployees + "\n";
      }

}
